### ISSUE TITLE
Support for sortedmulti_a

### ISSUE THREAD

**Issue by dr-orlovsky** [rust-bitcoin/rust-miniscript#320]
The support is part of Bitcoin core, but absent here

**Comment by sanket1729** [comment]
Note that this would only be supported at the top level. Supporting it at the miniscript level is not possible and is currently not in spec

**Comment by notmandatory** [comment]
Is it possible to remove this issue from the 8.0.0 milestone so that the next release can go out sooner? Projects like `bdk` need #450 before we can update to `rust-bitcoin` 0.29.

**Comment by apoelstra** [comment]
I would also like #478 -- let's bump this to the next milestone so we can do a quick release. Ok by you @sanket1729 ?

**Comment by apoelstra** [comment]
Oh, actually I don't care (I'm working on elements-miniscript but it is just a fork off the git repo, it doesn't use crates.io versions). But I agree with the sentiment that we ought to have a current release that matches rust-bitcoin.

**Comment by sanket1729** [comment]
In future releases, we should automatically release a new version of `rust-miniscript` whenever there is a new version of `rust-bitcoin`. Other normal feature releases of rust-miniscript can go at their own pace.

**Comment by sanket1729** [comment]
@apoelstra, I think I made some progress on changelog

**Comment by sanket1729** [comment]
Let me check my local branh

**Comment by sanket1729** [comment]
@apoelstra, there is already a changelog at https://github.com/rust-bitcoin/rust-miniscript/pull/462/commits/6f3c9672650444a358b7261be9a6b997902861ce

**Comment by sanket1729** [comment]
Adding top-level support for `sortedmulti_a` is certainly something we want to do. But something I don't have the time for now. Perhaps, a simple good first issue for someone who is looking for intermediate coding.

**Comment by ameya-deshmukh** [comment]
@sanket1729 I'd love to pick this up! Can I?

**Comment by sanket1729** [comment]
Sure, feel free to ask any questions here

**Comment by Harshit933** [comment]
Hello @sanket1729 @apoelstra I would like to take up this issue, Although I am still learning about descriptors, this would be a good issue to apply my knowledge.

**Comment by apoelstra** [comment]
@Harshit933 yep, this looks like a fairly self-contained issue. You can look at the existing `tr` descriptor for how Taproot works, and the existing `sortedmulti` descriptor to see what a top-level "sorted multisig" descriptor looks like.

**Comment by Harshit933** [comment]
> Only Taproot has the `CHECKSIGADD` opcode. If you are using the old `CHECKMULTISIG` then the existing `sortedmulti` will do you.

Is there any website where I can find all the opcode? 

> (Output) descriptors describe the script part of Bitcoin outputs (the other part of an output is a BTC amount, which stands on its own). They indicate whether the output is Taproot, segwit v0, P2SH, P2WPKH, etc etc. For descriptor types that have a freeform script (p2sh, p2wsh, taproot, etc) Miniscript sits inside and describes the script itself.

Thanks, this clears a lot of things.

**Comment by sanket1729** [comment]
@Harshit933, you can cross reference bitcoin core implementation for sortedmulti_a, while attempting this PR. That should serve as a good starting point for what needs to be done.
